gvr-modelviewer2/modelviewer2/src/main/java/org/gearvrf/modelviewer2/ModelViewer2Activity.java

```diff
@@ -0,0 +1,104 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.modelviewer2;
+
+import org.gearvrf.util.VRTouchPadGestureDetector;
+import org.gearvrf.util.VRTouchPadGestureDetector.OnTouchPadGestureListener;
+import org.gearvrf.util.VRTouchPadGestureDetector.SwipeDirection;
+import org.gearvrf.GVRActivity;
+import org.gearvrf.widgetplugin.GVRWidgetPlugin;
+
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.view.MotionEvent;
+import android.util.Log;
+
+public class ModelViewer2Activity extends GVRActivity implements
+        OnTouchPadGestureListener {
+
+    private GVRWidgetPlugin mPlugin = new GVRWidgetPlugin(this);
+    private VRTouchPadGestureDetector mDetector = null;
+    private ModelViewer2Manager mManager = null;
+    MyMenu mWidget;
+
+    @Override
+    protected void onCreate(Bundle icicle) {
+        super.onCreate(icicle);
+
+        DisplayMetrics displaymetrics = new DisplayMetrics();
+        getWindowManager().getDefaultDisplay().getMetrics(displaymetrics);
+        mPlugin.setViewSize(displaymetrics.widthPixels,
+                displaymetrics.heightPixels);
+
+        mDetector = new VRTouchPadGestureDetector(this);
+        mWidget = new MyMenu();
+
+        //SkyBox List
+        mManager = new ModelViewer2Manager(this, mPlugin);
+        mPlugin.setCurrentScript(mManager);
+        mWidget.mManager = mManager;
+
+        setScript(mManager, "gvr.xml");
+    }
+
+    @Override
+    public boolean onSingleTap(MotionEvent e) {
```

Comment:
The tap/press/swipe events should be processed after the GVRMain.onInit has been called. I have seen NPEs in case you tap immediately after the app has been launched.

```
08-10 15:08:02.208  6808  6808 E AndroidRuntime: FATAL EXCEPTION: main
08-10 15:08:02.208  6808  6808 E AndroidRuntime: Process: org.gearvrf.ModelViewer2Activity, PID: 6808
08-10 15:08:02.208  6808  6808 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'org.gearvrf.GVRScene org.gearvrf.GVRContext.getMainScene()' on a null object reference
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.modelviewer2.ModelViewer2Manager.getEyePointee(ModelViewer2Manager.java:203)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.modelviewer2.ModelViewer2Manager.onScroll(ModelViewer2Manager.java:311)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.modelviewer2.ModelViewer2Activity.onScroll(ModelViewer2Activity.java:79)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.util.VRTouchPadGestureDetector.onScroll(VRTouchPadGestureDetector.java:227)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:622)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.util.VRTouchPadGestureDetector.onTouchEvent(VRTouchPadGestureDetector.java:252)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at org.gearvrf.modelviewer2.ModelViewer2Activity.dispatchTouchEvent(ModelViewer2Activity.java:85)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at com.oculus.vrapi.InputHooks.dispatchTouchEvent(InputHooks.java:227)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2792)
08-10 15:08:02.208  6808  6808 E AndroidRuntime: 	at android.view.View.dispatchPointerEvent(View.java:10228)
```